aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -1545,19 +1541,23 @@ internal class RealAWSCognitoAuthPlugin(
                         val hubEvent = when {
                             authNState is AuthenticationState.SignedOut &&
                                 authZState is AuthorizationState.Configured -> {
-                                AuthChannelEventName.SIGNED_OUT
+                                AuthChannelEventName.SIGNED_OUT.toString()
                             }
                             authNState is AuthenticationState.SignedIn &&
                                 authZState is AuthorizationState.SessionEstablished -> {
-                                AuthChannelEventName.SIGNED_IN
+                                AuthChannelEventName.SIGNED_IN.toString()
+                            }
+                            authNState is AuthenticationState.FederatedToIdentityPool &&
+                                authZState is AuthorizationState.SessionEstablished -> {
+                                AWSCognitoAuthChannelEventName.FEDERATED_TO_IDENTITY_POOL.toString()
                             }
                             deleteUserAuthZState?.deleteUserState is DeleteUserState.UserDeleted -> {
-                                AuthChannelEventName.USER_DELETED
+                                AuthChannelEventName.USER_DELETED.toString()
```

Comment:
`HubEvent.create` does call `.toString` when creating the event. Do we need to convert to string here?

Reply:
Had to move all to toString because of the new AWSCognitoAuthChannelEventName.FEDERATED_TO_IDENTITY_POOL. the return type for events had to be a string, instead of AuthChannelEventName